docs/adr/0008-kommunikasjon-mellom-sbs-og-regel-api.md

```diff
@@ -0,0 +1,48 @@
+# Kommunikasjon mellom applikasjon i sbs og regel-api 
+
+* Status: pending
+* Deciders: Team Dagpenger
+* Date: 2019-10-06 
+
+## Context and Problem Statement
+
+Vår nye dagpengekalkulator er en front-end som skal leve i sbs. Den trenger å snakke med dp-regel-api i fss.
+I fremtiden vil vi også få flere ting som lever i sbs, så vi må finne en fornuftig måte å kommunisere 
+
+## Considered Options
+
+* Sonekryssing 
+* Deploye regel-api i begge soner
+* Bruke kafka som regel-api-endepunkt 
+* … 
+
+## Pros and Cons of the Options 
+
+### Sonekryssing 
+
+#### Cons
+
+- Det er komplisert å sette opp. Riktignok en one time job: Se https://confluence.adeo.no/display/AR/Sonekrysning
+
+### Deploye regel-api i begge soner
+
+#### Cons
+- Mere å monitorere
+- Credentials til DB må legges inn i Vault
```

Comment:
Blir ikke dette vanskelig med flyway?

Reply:
Hvordan da? Ser på det som 4 podder hvor to kjører i hvert sitt kluster

Reply:
Nå er det vel sånn at migrate gjøres i startup av appen? Så når k8s deployer dette starter den en ny pod og venter på isReady før den starter neste? Med andre ord, en pod får lov å migrere i ro og orden før neste pod kommer opp.

Det funker jo fint når det er samme ReplicaSet, men om vi får to ulike Deployments så vil det jo potensielt bli to pods som prøver å migrere samtidig.

Reply:
Jeg tror det er flyway som sørger for synkroniseringen via et flagg eller noe i databasen.

Reply:
Hvis ikke må vi lage en PR til Flyway for det er jo bare default verdier som gjør at det er en og en pod som starter opp :) . Antall nye pods som starter opp samtidig ved en ny deployment kan jo styres via "max surge."